be/src/storage/inverted/clucene/clucene_inverted_reader.cpp

```diff
@@ -111,6 +109,7 @@ Status FullTextCLuceneInvertedReader::query(OlapReaderStatistics* stats, const s
         return Status::InternalError(fmt::format("CLuceneError occured, error msg: {}", e.what()));
     }
     bit_map->swap(result);
+    CLOSE_DIR(directory)
```

Comment:
if match_operator->match return error, the directory will not be closed?

Reply:
Fixed it, add additional CLOSE_DIR(directory) when match is fail

Reply:
Better using DeferOp ?

Reply:
if return Status::InvalidArgument("Unknown query type");